services/tradingService.js

```diff
@@ -44,35 +48,57 @@ const trade = async (tradeData) => {
 
     switch (tradeType) {
       case 'SELL': {
+        if (!userStocks.id || userStocks.quantity < quantity) {
+          return { canUserTrade: false, errorMessage: INSUFFICIENT_QUANTITIES }
+        }
+
         quantityToUpdate = quantity + stockData.quantity
-        userBalance = quantity * stockData.price
-        updatedCurrencyData.dinero = userBalance + currencies.dinero
-        stockPriceToBeUpdated = getUpdatedPrice(stockData.price)
+        userBalance = (quantity * stockData.price) + currencies[walletConstants.DINERO]
+        userStocksQty = userStocks.quantity - quantity
         break
       }
       case 'BUY': {
         qtyUserCanPurchase = Math.floor(totalPrice / stockData.price)
-        if (qtyUserCanPurchase <= 0 || totalPrice > currencies.dinero) {
+        if (qtyUserCanPurchase <= 0 || totalPrice > currencies[walletConstants.DINERO]) {
```

Comment:
I see the wallet constants being used a lot of times in this file. Can we destructure them and use?

Reply:
a383c4b